recipe/meta.yaml

```
@@ -91,23 +89,17 @@ outputs:
         - setuptools_scm
         - wheel
         - numpy
-        # https://github.com/single-cell-data/TileDB-SOMA/issues/1926
-        - pyarrow <13.0 # [osx]
-        - pyarrow       # [not osx]
-        - pyarrow-hotfix
+        - pyarrow
       run:
         - {{ pin_compatible('numpy', lower_bound='1.16', upper_bound='1.27') }}
         - {{ pin_subpackage('libtiledbsoma', exact=True) }}
         - pandas
-        # https://github.com/single-cell-data/TileDB-SOMA/issues/1926
-        - pyarrow <13.0 # [osx]
-        - pyarrow       # [not osx]
-        - pyarrow-hotfix
+        - pyarrow
```

If we are removing the fallback of pyarrow-hotfix, I think we need to set the lower bound to `>=14` (for `build`, `host`, and `run`)

```suggestion
        - pyarrow >=14
```

---

https://github.com/TileDB-Inc/tiledbsoma-feedstock/pull/171/commits/a972b967d7e3f58f7ecaf58e1080b5a0abadb992

Thanks @jdblischak !